records/generator.go

```diff
@@ -21,12 +23,15 @@ import (
 // Will likely become map[string][]discoveryinfo
 type rrs map[string][]string
 
+var defaultDomainPatterns = []string{"{{.tname}}"}
+
 // Mesos-DNS state
 // Refactor when discovery id is available
 type RecordGenerator struct {
 	As     rrs
 	SRVs   rrs
 	Slaves map[string]string
+	DomainPatterns []string
```

Comment:
the current design passes Config items to each func as needed, why is this special? If RecordGenerator is going to house Config items, then perhaps it should declare Config as an anonymous field:
```go
type RecordGenerator struct {
  Config
  As rrs
  SRVs rrs
  //...
}
```

this allows:

1. simple Config assignment: `rg.Config = someOtherConfig`
1. simple access to embedded Config field's from RecordGenerator obj: `for _, pattern in rg.DomainPatterns {`
1. a more consistent initialization of default config values (`defaultDomainPatterns` probably belongs in Config with some of the other defaults)